packages/appium/docs/en/guides/caps.md

```diff
@@ -2,9 +2,150 @@
+capabilities (otherwise it will not know what app to install and/or launch). Each driver will
+document how it interprets these capabilities and any other platform-specific requirements.
+
+## Constructing capabilities using clients
+
+Each Appium client has its own way of constructing capabilities and starting a session. Here is
+a minimal example of starting an Appium Android session using different client libraries.
+
+=== "JS (WebDriverIO)"
+
+    ```js
+    const {remote} = require('webdriverio')
+
+    const capabilities = {
+      platformName: 'Android',
+      'appium:automationName': 'UiAutomator2',
+      'appium:deviceName': 'Android',
+      'appium:appPackage': 'com.android.settings',
+      'appium:appActivity': '.Settings',
+    }
+
+    const wdOpts = {
+      host: 'localhost',
+      port: 4723,
+      capabilities,
+    }
+
+    const driver = await remote(wdOpts)
+    ```
+
+=== "Java"
+
+    ```java
+    /* TODO */
+    ```
+
+=== "Python"
+
+    ```py
+    from appium import webdriver
+
+    capabilities = {
+      'platformName': 'Android',
+      'automationName': 'UiAutomator2',
+      'deviceName': 'Android',
+      'appPackage': 'com.android.settings',
+      'appActivity': '.Settings',
+    }
+
+    driver = webdriver.Remote(command_executor='http://localhost:4723',
+                              desired_capabilities=capabilities)
+    ```
+
+=== "Ruby"
+
+    ```rb
+    # TODO
+    ```
+
+=== "C#"
+
+    ```dotnet
+    /* TODO */
+    ```
+
+## Always-match and first-match capabilities
+
+The W3C spec allows clients to give the Appium server some flexibility in the kind of session it
+creates in response to a new session request. This is through the concept of "always-match" and
+"first-match" capabilities:
+
+- Always-match capabilities consist of a single set of capabilities, every member of which must be satisfied by the server in order for the new session request to proceed.
+- First-match capabilities consist of an array of capability sets. Each set is merged with the always-match capabilities, and the first set that the server knows how to handle will be the set that is used to start the session.
+
+In practice, knowledge or use of always-match or first-match capabilities are not necessary or
+recommended for use with Appium. Instead, we recommend that you simply define the exact set of
+capabilities you want the Appium server to handle. These will be encoded as the always-match
+capabilities, and the array of first-match capabilities will be empty.
+
+That being said, the Appium server does process always-match and first-match capabilities as
+defined in the W3C spec, so if you use these features, Appium will work as expected. The process of
+defining always-match and first-match capabilities is unique to each client library, so refer to
+the documentation for your client library to see examples of how it works.
+
+## Special notes for cloud providers
+
+!!! warning
```

Comment:
what is this markdown

Reply:
it's not designed for rendering on github. it's part of mkdocs-material